staging/src/k8s.io/csi-translation-lib/plugins/in_tree_volume.go

```diff
@@ -165,6 +218,153 @@ func translateTopology(pv *v1.PersistentVolume, topologyKey string) error {
 	return nil
 }
 
+// getTopologyLabel checks if the kubernetes topology used in this PV is GA
+// and return the zone/region label used.
+// It first check the NodeAffinity to find topology. If nothing is found,
+// It checks the PV labels. If it is empty in both places, it will return
+// GA label by default
+func getTopologyLabel(pv *v1.PersistentVolume) (bool, string, string) {
+
+	// Check the NodeAffinity first for the topology version
+	zoneGA := TopologyKeyExist(v1.LabelTopologyZone, pv.Spec.NodeAffinity)
+	regionGA := TopologyKeyExist(v1.LabelTopologyRegion, pv.Spec.NodeAffinity)
+	if zoneGA || regionGA {
```

Comment:
Yeah, firstly it will be very unexpected... But it could happen, my criteria here is that if it has a GA version topology, then it should be treated as GA supported.

Reply:
I guess what if zone label is beta, but region label is GA. Then we will fail to fine the zone because we're looking for GA zone label?

Reply:
Hmm, so do you suggest we return zone label version and region label version separately? Like we can return GA version region label and beta version zone label in the case you described?

Reply:
That's a good idea. I think it will simplify the logic too

Reply:
As we discussed offline, it would be good to not overcomplicate this, so I will keep the old logic which will assume Zone and Region labels are in the same version.

Reply:
Sorry coming back to this again. Seeing `zone || region` makes me asks questions like "what if zone was false?". Since we care about zone most importantly, what if we only check zone, and then return the corresponding region version based off of what we found? (ie, if we found beta zone, then return beta region without checking)

Reply:
Done